index.js

```diff
@@ -17,7 +17,9 @@ function fastifyMysql (fastify, options, next) {
 
     const client = connectionType !== 'connection' ? db.pool : db.connection
 
-    fastify.addHook('onClose', (fastify, done) => client.end(done))
+    fastify.addHook('onClose', async (fastify, done) => {
+      await client.end(); done()
+    })
```

Comment:
I really unclear as to how this change is any different from the original.

Reply:
All I understood is that when `mysql` promise is true `client.end(done)` callback function is not valid and `client.end()` is an async function. If I revert it to the old one, the new tests fail. 

```text
test/closing.test.js .................................. 6/7
  promise connection
  not ok test unfinished
    stack: |
      Object.<anonymous> (test/closing.test.js:29:1)
      Generator.next (<anonymous>)
      internal/main/run_main_module.js:17:47
    test: promise connection
    at:
      line: 29
      column: 1
      file: test/closing.test.js
      function: Object.<anonymous>
    source: |
      test('promise connection', (t) => {

test/connection.test.js ............................. 38/38
test/initialization.test.js ......................... 19/19
test/pool.promise.test.js ........................... 13/13
test/pool.test.js ................................... 35/35
total ............................................. 111/112
  

  111 passing (3s)
  1 failing
```

and with the new code
```text
> fastify-mysql@1.0.0 test /Users/Jerry/Code/issues/fastify-mysql
> standard && tap test/*.test.js

test/closing.test.js .................................. 7/7
test/connection.test.js ............................. 38/38
test/initialization.test.js ......................... 19/19
test/pool.promise.test.js ........................... 13/13
test/pool.test.js ................................... 35/35
total ............................................. 112/112

  112 passing (3s)

  ok
```

Reply:
When we use `require('mysql2/promise')`, `client.end()` return every time a Promise, even if a callback is provided

https://github.com/sidorares/node-mysql2/blob/master/promise.js#L64

Reply:
I think it would be better to use the `usePromise` boolean to determine if it should be:

```js
client.end(done)
```

or

```js
client.end().then(done).catch(done)
```